tensorflow/java/src/main/native/saved_model_bundle_jni.cc

```diff
@@ -20,14 +20,36 @@ limitations under the License.
 #include "tensorflow/java/src/main/native/exception_jni.h"
 #include "tensorflow/java/src/main/native/saved_model_bundle_jni.h"
 
+
 JNIEXPORT jobject JNICALL Java_org_tensorflow_SavedModelBundle_load(
     JNIEnv* env, jclass clazz, jstring export_dir, jobjectArray tags,
-    jbyteArray run_options) {
+    jbyteArray run_options, jbyteArray session_options) {
   TF_Status* status = TF_NewStatus();
   jobject bundle = nullptr;
 
   // allocate parameters for TF_LoadSessionFromSavedModel
   TF_SessionOptions* opts = TF_NewSessionOptions();
+
+
+  TF_Buffer* csession_options = nullptr;
+  if (session_options != nullptr) {
+	  size_t sz = env->GetArrayLength(session_options);
+	  if(sz > 0){
+		  jbyte* sconfigbyte = env->GetByteArrayElements(session_options, nullptr);
+		  csession_options =
+		           TF_NewBufferFromString(static_cast<void*>(sconfigbyte), sz);
+		  env->ReleaseByteArrayElements(session_options, sconfigbyte, JNI_ABORT);
+	      TF_SetConfig(opts, csession_options->data,
+					  	  	  	  	  csession_options->length, status);
+	      if (!throwExceptionIfNotOK(env, status)) {
+	    	   	   TF_DeleteSessionOptions(opts);
+	           TF_DeleteStatus(status);
+	           TF_DeleteBuffer(csession_options);
```

Comment:
You should be able to unconditionally call `TF_DeleteBuffer` after calling `TF_SetConfig`.

Reply:
I apologize, I didn't quite follow your comment.
I was suggesting that instead of calling `TF_DeleteBuffer(csession_options)` in two places (once inside the `if (!throwExceptionIfNotOK(...` block and once in line 87, we could call it immediately after  invoking `TF_SetConfig`.